xwiki-rendering-api/src/main/java/org/xwiki/rendering/internal/parser/reference/GenericLinkReferenceParser.java

```diff
@@ -263,19 +263,16 @@ protected String parseElementAfterString(StringBuilder content, String separator
 
         // Find the first non escaped separator (starting from the end of the content buffer).
         int index = content.lastIndexOf(separator);
-        while (index != -1) {
+        while (index > 0) {
```

Comment:
This will break the use case where the separator is the first character of the string.